app/src/main/java/com/github/polypoly/app/map/MapActivity.kt

```diff
@@ -360,42 +415,74 @@ class MapActivity : ComponentActivity() {
         return BitmapDrawable(context.resources, scaledBitmap)
     }
 
+    /**
+     * Initializes the location overlay and sets the location listener.
+     */
     private fun initLocationOverlay(mapView: MapView): MyLocationNewOverlay {
         val locationProvider = GpsMyLocationProvider(mapView.context)
         var lastLocation = Location("")
 
         val locationOverlay = object : MyLocationNewOverlay(locationProvider, mapView) {
             override fun onLocationChanged(location: Location?, provider: IMyLocationProvider?) {
                 super.onLocationChanged(location, provider)
-                updateAllDistances(mapView, GeoPoint(location))
+                mapViewModel.setCloseLocation(
+                    updateAllDistancesAndClosest(mapView, GeoPoint(location))
+                )
                 mapViewModel.addDistanceWalked(lastLocation.distanceTo(location!!))
                 lastLocation = locationProvider.lastKnownLocation
+                print("location: $location")
```

Comment:
Is it a forgotten debug print?

Reply:
I write so little debug prints in this project due to the nature of the bugs that i forgot that one, thanks!